app/src/main/java/com/concordium/wallet/ui/more/notifications/NotificationsPreferencesActivity.kt

```diff
@@ -0,0 +1,114 @@
+package com.concordium.wallet.ui.more.notifications
+
+import android.Manifest
+import android.os.Build
+import android.os.Bundle
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.activity.result.registerForActivityResult
+import androidx.annotation.RequiresApi
+import androidx.core.view.isVisible
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.get
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import com.concordium.wallet.R
+import com.concordium.wallet.databinding.ActivityNotificationsPreferencesBinding
+import com.concordium.wallet.ui.base.BaseActivity
+import com.concordium.wallet.util.Log
+import kotlinx.coroutines.launch
+
+class NotificationsPreferencesActivity : BaseActivity(
+    R.layout.activity_notifications_preferences,
+    R.string.notifications_preferences_title,
+) {
+    private val binding: ActivityNotificationsPreferencesBinding by lazy {
+        ActivityNotificationsPreferencesBinding.bind(findViewById(R.id.root_layout))
+    }
+    private val viewModel: NotificationsPreferencesViewModel by lazy {
+        ViewModelProvider(
+            this,
+            ViewModelProvider.AndroidViewModelFactory.getInstance(application)
+        ).get()
+    }
+
+    @RequiresApi(Build.VERSION_CODES.TIRAMISU)
+    private val notificationPermissionLauncher =
+        registerForActivityResult(
+            ActivityResultContracts.RequestPermission(),
+            Manifest.permission.POST_NOTIFICATIONS,
+            this::onNotificationPermissionResult
+        )
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        initViewModel()
+        initViews()
+
+        hideActionBarBack(isVisible = true)
+    }
+
+    private fun initViewModel() {
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                launch {
+                    viewModel.areCcdTxNotificationsEnabledFlow.collect {
```

Comment:
We have `collectWhenStarted` extension to avoid such nesting, please, use it instead.